py_zipkin/logging_helper.py

```diff
@@ -343,6 +344,10 @@ def __exit__(self, _exc_type, _exc_value, _exc_traceback):
         else:
             self.flush()
 
+    def _reset_queue(self):
+        self.queue = []
+        self.current_size = 5  # Thrift list size
```

Comment:
Can you pull this magic number out into an ALL_CAPS_CONSTANT

Also can you change the comment to say that this is the thrift list header/overhead, not the size of the list itself, and that the size is in bytes 

I'm also a little bit concerned that we are muddling the concerns of thrift encoding by splitting it apart into two stages (the span encoding and the list construction). I wonder if we can think a little about a cleaner way to do this. Maybe something like subclassing the queue and having `__len__` return the adjusted size

Reply:
> Maybe something like subclassing the queue and having __len__ return the adjusted size

That won't help with doing add_span in 2 stages since we need to check whether the size would go over the limit if we added the current span.

It'd only simplify a bit the if, but that doesn't seem a particularly big improvement to me

Reply:
I could do https://i.fluffy.cc/GkFcHZddhX11V1C2JPTXPxQ8CjcRrS00.html, but that doesn't seem a great improvement.

Or maybe it is since then the JSONBatchSender could reuse it?